### PR TITLE
Expose LRU cache options

### PR DESCRIPTION
**Pull request by athre0z**
This PR exposes the following two DB options related to BlobDB caching:

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/include/rocksdb/advanced_options.h#L1129-L1148

### CODE REVIEW

**Comment by athre0z** [comment]
Added one more commit to also expose the advanced LRU cache constructor with custom options.

File: src/db_options.rs [hunk 1]
```diff
@@ -57,6 +57,13 @@ impl Cache {
         Cache(Arc::new(CacheWrapper { inner }))
     }
 
+    /// Creates an LRU cache with custom options.
+    pub fn new_lru_cache_opts(opts: &LruCacheOptions) -> Cache {
+        let inner =
+            NonNull::new(unsafe { ffi::rocksdb_cache_create_lru_opts(opts.inner) }).unwrap();
```
Comment (stanislav-tkach) [inline src/db_options.rs:63]: Am I missing something or is there an issue in the code below?
```rust
let options = LruCacheOptions::default();
let cache = Cache::new_lru_cache_opts(&options);
drop(options);
// Now the cache holds a pointer to the freed memory.
```

Comment (athre0z) [reply src/db_options.rs:63]: The pointer to the options is not stored in the resulting cache object; it is only borrowed for the duration of construction. See the following call chain:

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/db/c.cc#L4825

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/include/rocksdb/cache.h#L286

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/cache/lru_cache.cc#L689

https://github.com/facebook/rocksdb/blob/f6fd4b9dbd15dba36f7e5ad23de407b5c26b1460/cache/lru_cache.cc#L651C43-L651C47

Comment (stanislav-tkach) [reply src/db_options.rs:63]: Thanks for the clarification!

**Review by stanislav-tkach — APPROVED** [review]

**Comment by zaidoon1** [comment]
This also LGTM, adding tests would be great

**Comment by athre0z** [comment]
I can write some tests, but it might take a while until I get to it depending on how deep these tests should go. Simply adding tests that invoke the new methods to make sure it links & doesn't segfault is quickly done, but I might have to read up a bit deeper if you'd like tests that actually make sure that RocksDB is caching things as expected etc (like some other test cases seem to do). What's the expectation here?

**Comment by zaidoon1** [comment]
> I can write some tests, but it might take a while until I get to it depending on how deep these tests should go: I'm unfortunately rather short on time right now. Simply adding tests that invoke the new methods to make sure it links & doesn't segfault is quickly done, but I might have to read up a bit deeper if you'd like tests that actually make sure that RocksDB is caching things as expected etc (like some other test cases seem to do). What's the expectation here?

a simple test case where you set the new options being exposed here, and put some kvs without segfaulting is good enough

**Comment by zaidoon1** [comment]
LGTM, will merge once builds pass

**Review by zaidoon1 — APPROVED** [review]

**Pull request merged**